src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java

```diff
@@ -320,6 +325,40 @@ private boolean isAnotherBuildWithSameParametersRunningOnAnyNode(Queue.Item item
         return false;
     }
 
+    private boolean isThrottleBuildInProgress(Queue.Item item) {
+        ThrottleJobProperty tjp = getThrottleJobProperty(item.task);
+        if (tjp == null) {
+            // If the property has been occasionally deleted by this call,
+            // it does not make sense to limit the throttling by parameter.
+            return false;
+        }
+
+        List<String> paramsToCompare = tjp.getParamsToCompare();
+        List<ParameterValue> itemParams = getParametersFromQueueItem(item);
+
+        if (paramsToCompare.size() > 0) {
+            itemParams = doFilterParams(paramsToCompare, itemParams);
+        }
+
+        if (item.task instanceof Job) {
+            Job<?, ?> project = (Job<?, ?>) item.task;
+            for (Run<?, ?> run : project.getBuilds()) {
```

Comment:
:bug: This will load every build of a project from disk, which is a huge performance hit and not appropriate for a hot method like this. Use `FlowExecutionList.get` instead to get active Pipeline jobs in memory. See also `ExecutorStepExecution#getOwnerExecutable`.